pkg/health/server/prober.go

```diff
@@ -283,45 +361,24 @@ func (p *prober) httpProbe(node string, ip string) *models.ConnectivityStatus {
 		if debugLogsEnabled {
 			scopedLog.WithField("rtt", rtt).Debug("Greeting successful")
 		}
-
 		result.Status = ""
 		result.Latency = rtt.Nanoseconds()
 	} else {
 		if debugLogsEnabled {
 			scopedLog.WithError(err).Debug("Greeting failed")
 		}
-
 		result.Status = err.Error()
 	}
 
-	return result
+	resChan <- connectivityResult{ip: ip, status: result}
 }
 
-func (p *prober) getIPsByNode() map[string][]*net.IPAddr {
-	p.RLock()
-	defer p.RUnlock()
-
-	// p.nodes is mapped from all known IPs -> nodes in N:M configuration,
-	// so multiple IPs could refer to the same node. To ensure we only
-	// ping each node once, deduplicate nodes into map of nodeName -> []IP.
-	nodes := make(map[string][]*net.IPAddr)
-	for _, node := range p.nodes {
-		if nodes[node.Name] != nil {
-			// Already handled this node.
-			continue
-		}
-		nodes[node.Name] = []*net.IPAddr{}
-		for elem, primary := range node.Addresses() {
-			if _, addr := resolveIP(&node, elem, primary); addr != nil {
-				nodes[node.Name] = append(nodes[node.Name], addr)
-			}
-		}
-	}
-
-	return nodes
-}
+func (p *prober) runProbe() {
```

Comment:
Ah oops yes I meant the synchronous prober! 
Totally on the same page about deprecating the synchronous prober. Especially for larger clusters it would defeat the purpose of this change (and the probing interval change upcoming) => so then, to confirm:
- Currently, synchronous prober is returning the results of the most recent probe. I would need to update the docs to mention this behavior change?
- Maybe, as a separate PR, we would want to change the CLI connectivity tests to use `cilium-health status` instead of `cilium-health status --probe` and delete the ability to `--probe`? My impression was that --probe isn't recommended for use, so the deprecation process could be like this. But I'm not sure if it should be more formal, with a deprecation warning before removal like metrics?

Reply:
We can mark `--probe` option as hidden in case someone has some automation that uses it so we don't break them. 
We can also add update note saying that --probe is no longer supported and returns cached results instead.

Reply:
Got it, this makes sense! :) Thank you!